Write token to a temporary directory

jakirkham opened the pull request:
Simplifies the cleanup steps involved with writing and getting rid of the token. Also has the bonus of being somewhere that is writable given the lock down of Docker permissions. ( https://github.com/conda-forge/docker-images/pull/39 )

jakirkham committed:
MNT: Re-rendered with conda-smithy 2.4.5

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Release 4.4.15

Make a patch release to handle the changes to where the token is
written.

jakirkham committed:
Write token to a temporary file

This allows it to be cleaned up automatically after being written. Also
has the bonus of being somewhere we can write. Currently writing to the
recipe directory and feedstock are out given the locking down of
permissions in the Docker image.

jakirkham commented:
Only care about Linux here. So went ahead and merged. After all if this is broken, it will actually not release.

jakirkham commented:
Seems to have released fine on Linux, which means the other feedstocks should be fine to release now too.

jakirkham commented:
Were some issues on Windows. Trying to correct them in PR ( https://github.com/conda-forge/conda-forge-build-setup-feedstock/pull/86 ).
